doc/source/programs/gdal_raster_tile.rst

```diff
@@ -0,0 +1,236 @@
+
+    - ``CBMTILE``: Lambert Conformal Conic-based tiled coordinate reference system for Canada. See `definition <https://github.com/OSGeo/gdal/blob/master/gcore/data/tms_MapML_CBMTILE.json>`__
+
+    - ``NZTM2000``: LINZ NZTM2000 Map Tile Grid. see `definition <https://github.com/OSGeo/gdal/blob/master/gcore/data/tms_NZTM2000.json>`__
+
+    Additional tiling schemes are discovered from tms_XXXX.json files placed in GDAL data directory.
+
+.. option:: --min-zoom <MIN-ZOOM>
+
+    Minimum zoom level to generate. If not specified, equal to :option:`--max-zoom`.
+
+.. option:: --max-zoom <MIN-ZOOM>
+
+    Maximum zoom level to generate. If not specified, this will be determined by
+    comparing the resolution of the input dataset with the closest resolution in
+    the list of tile matrix of the tile matrix set.
+
+.. option:: --min-x <MIN-X>
+
+    Minimum tile X coordinate, at maximum zoom level. Can be set to restrict
+    the tiling process.
+
+.. option:: --min-y <MIN-Y>
+
+    Minimum tile Y coordinate, at maximum zoom level. Can be set to restrict
+    the tiling process.
+
+.. option:: --max-x <MAX-X>
+
+    Maximum tile X coordinate, at maximum zoom level. Can be set to restrict
+    the tiling process.
+
+.. option:: --max-y <MAX-Y>
+
+    Maximum tile Y coordinate, at maximum zoom level. Can be set to restrict
+    the tiling process.
+
+.. option:: --no-intersection-ok
+
+    Whether the dataset extent not intersecting the tile matrix is only a warning.
+    Otherwise, by default, an error will be emitted if that occurs.
+
+.. option:: -r, --resampling nearest|bilinear|cubic|cubicspline|lanczos|average|rms|mode|min|max|med|q1|q3|sum
+
+    Resampling method used to generate maximum zoom level, and also lower zoom
+    levels if :option:`--overview-resampling` is not specified.
+    Defaults to ``cubic``.
+
+.. option:: --overview-resampling nearest|bilinear|cubic|cubicspline|lanczos|average|rms|mode|min|max|med|q1|q3|sum
+
+    Resampling method used to generate zoom levels lower than the maximum zoom
+    level. Defaults to the value of :option:`--resampling` is not specified.
+
+.. option:: --convention xyz|tms
+
+    Tile numbering convention:
+
+    - ``xyz`` (default): from top, as in OGC Web Map Tiling Specification (WMTS)
+
+    - ``tms``: from bottom, as in OSGeo Tile Map Service (TMS) Specification.
+
+.. option:: --tilesize <PIXELS>
+
+    Width and height in pixel of a tile. Default is 256 for default tiling schemes.
+    Setting it to a higher value enables to generate higher DPI tile sets.
+
+.. option:: --addalpha
+
+    Whether to force adding an alpha channel. An alpha channel is added by default,
+    unless the source dataset has a nodata value and the output format supports its.
+
+.. option:: --noalpha
+
+    Disable the creation of an alpha channel.
+
+.. option:: --dstnodata <DSTNODATA>
```

Comment:
Would be good to consistently use a hyphen (or not) for `--dst` and `--src` options. I see we have `--dst-crs` elsewhere but `--dstnodata` here (and elsewhere). Also `--dstmin` and `--dstmax`.

There's also a question of `--src` and `--dst` vs `--input` and `--output`...

Reply:
yes, we could do better. Any preference ? We should move that discussion to a dedicated ticket

Reply:
Personal preference is for the hyphen. Also `--tile-size`, `--add-alpha`, `--no-alpha`